src/SecurityBundle/src/CmsUser/CmsUserModel.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+namespace ChameleonSystem\SecurityBundle\CmsUser;
+
+use Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+class CmsUserModel implements UserInterface, PasswordAuthenticatedUserInterface
+{
+    public function __construct(
+        readonly private \DateTimeImmutable $dateModified,
+        readonly private string $id,
+        readonly private string $userIdentifier,
+        readonly private string $firstname,
+        readonly private string $lastname,
+        readonly private string $company,
+        readonly private string $email,
+        readonly private string $cmsLanguageId,
+        readonly private array $availableLanguagesIsoCodes,
+        readonly private ?string $currentEditLanguageIsoCode,
+        readonly private array $availableEditLanguages,
+        readonly private ?string $password = null,
+        readonly private array $roles = [],
+        readonly private array $rights = [],
+        readonly private array $groups = [],
+        readonly private array $portals = []
```

Comment:
Why make those private with getters, if they could be public? They are readonly.

Reply:
not sure I like to lose my getter...

Reply:
You do ;)

Reply:
If you know how to add the proper code hint, then yes. Otherwise, I prefer getters for now

Reply:
see above

Reply:
see above :)